Packages/vcs/Lib/vcsvtk/fillareautils.py

```diff
@@ -33,8 +34,14 @@ def make_patterned_polydata(inputContours, fillareastyle=None,
     # Scaled the size to 2 times to make the pattern image of a finer resolution
     xBounds = bounds[1] - bounds[0]
     yBounds = bounds[3] - bounds[2]
-    xres = int(4.0*xBounds)
-    yres = int(4.0*yBounds)
+    if xBounds <= 1 and yBounds <= 1 and size is not None:
+        xBounds *= size[0]
```

Comment:
@chaosphere2112 Perhaps you could use the Normalized Viewport coordinates instead of render window size?